chat-plugins/calculator.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+function isNumeric(str) {
+	return !isNaN(parseFloat(str));
+}
+
+function parseMathematicalExpression(infix) {
+	// Shunting-yard Algorithm -- https://en.wikipedia.org/wiki/Shunting-yard_algorithm
+	let outputQueue = [];
+	let operatorStack = [];
+	let operators = {
+		"^": {
+			precedence: 4,
+			associativity: "Right",
+		},
+		"/": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"*": {
+			precedence: 3,
+			associativity: "Left",
+		},
+		"+": {
+			precedence: 2,
+			associativity: "Left",
+		},
+		"-": {
+			precedence: 2,
+			associativity: "Left",
+		},
+	};
+	infix = infix.replace(/\s+/g, "");
+	infix = infix.split(/([+\-*/^()])/);
+	infix = infix.filter(token => token);
+	for (const token of infix) {
+		if ("^*/+-".includes(token)) {
+			let op = operators[token];
+			let prevToken = operatorStack[operatorStack.length - 1];
+			let prevOp = operators[prevToken];
+			while ("^*/+-".includes(prevToken) && (
+				op.associativity === "Left" ? op.precedence <= prevOp.precedence : op.precedence < prevOp.precedence
+			)) {
+				outputQueue.push(operatorStack.pop());
+				prevToken = operatorStack[operatorStack.length - 1];
+				prevOp = operators[prevToken];
+			}
+			operatorStack.push(token);
+		} else if (token === "(") {
+			operatorStack.push(token);
+		} else if (token === ")") {
+			while (operatorStack[operatorStack.length - 1] !== "(") {
+				outputQueue.push(operatorStack.pop());
+			}
+			operatorStack.pop();
+		} else {
+			outputQueue.push(token);
+		}
+	}
+	while (operatorStack.length > 0) {
+		outputQueue.push(operatorStack.pop());
+	}
+	return outputQueue;
+}
+
+function solveRPN(rpn) {
+	let resultStack = [];
+	for (const token of rpn) {
+		if (isNumeric(token) === true) {
+			resultStack.push(token);
```

Comment:
Wait, the conversion should be done here.

```suggestion
			resultStack.push(Number(token));
```

Reply:
`Number` is better than `parseFloat` because it'll convert non-numbers to `NaN`.

Reply:
done.